### PR TITLE
Add `Task.seq` and `Task.forEach`

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
This was recently added to basic-cli in https://github.com/roc-lang/basic-cli/pull/156

### CODE REVIEW

**Comment by lukewilliamboswell** [comment]
Posted in [zulip](https://roc.zulipchat.com/#narrow/stream/316715-contributing/topic/Casual.20Conversation/near/416675309) 
> There is an issue [in CI](https://github.com/roc-lang/basic-webserver/actions/runs/7577479595/job/20638360851?pr=19) I haven't seen before -- it should be totally unrelated to the changes in that PR. The surgical linker is unhappy with `__umodti3`. I've had a quick look and haven't resolved yet, posting here in case anyone else is able to look at this further. I'll circle back later, but want to progress some other things rn.
```
thread 'main' panicked at 'Undefined Symbol in relocation, (+4ef2, Relocation { kind: PltRelative, encoding: Generic, size: +20, target: Symbol(SymbolIndex(+88)), addend: +fffffffffffffffc, implicit_addend: false }): Ok(Symbol { name: "__umodti3", address: +0, size: +0, kind: Unknown, section: Undefined, scope: Unknown, weak: false, flags: Elf { st_info: +10, st_other: +0 } })', crates/linker/src/elf.rs:1486:25
```

File: platform/Task.roc [hunk 1]
```diff
@@ -217,3 +219,35 @@ batch = \current -> \next ->
...
+## getAuthors : Task (List Author) [DbError]
+## getAuthors = Task.list authors getAuthor
+## ```
+##
+seq : List (Task ok err) -> Task (List ok) err
+seq = \tasks ->
+    List.walk tasks (Task.ok []) \state, task ->
+        value <- task |> Task.await
+
+        state |> Task.map \values -> List.append values value
```
Comment (RMailaender) [inline platform/Task.roc:238]: I think the comment doesn't fit the implementation. It seems to me, `Task.seq` actually converts a `List (Task ok err)` to a `Task (List ok) err`.

Or is there a `Task.list` function missing here?

**Review by RMailaender — CHANGES_REQUESTED** [review]
I think the description for `seq` doesn't fit.

File: platform/Task.roc (hunk 1 above)
Comment (lukewilliamboswell) [reply platform/Task.roc:264]: Thank you @RMailaender 

This implementation is copied from basic-cli. https://github.com/roc-lang/basic-cli/blob/307f1f91d984ac966cca75a40d403bdced1f05a6/platform/Task.roc#L227

I think if we are going to change this, we should discuss this further on zulip, and try to keep the API for basic-cli and basic-web server aligned.

**Review by Anton-4 — APPROVED** [review]

**Pull request merged**